pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -901,32 +1075,35 @@ func shouldCompressResponse(request, response *dns.Msg) bool {
 	return false
 }
 
-func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
-	newRE := make(CachedSelectorREEntry)
-	for selector, l7Rules := range l7 {
-		if l7Rules == nil {
-			l7Rules = &policy.PerSelectorPolicy{L7Rules: api.L7Rules{DNS: []api.PortRuleDNS{{MatchPattern: "*"}}}}
+// GenerateRegexpAndFqdns takes a set of l7Rules and returns a regex for matching the matchPatterns
+// and a list of the provided match names, all of them as FQDNs. In case a domain is present in the list of FQDNs
```

Comment:
```suggestion
// and a list of the provided matchNames, all of them as FQDNs. In case a domain is present in the list of FQDNs
```

More consistency with out terminology. Please also go through and ensure we are using consistent terms. This code is already quite abstract (not your fault :wink:) in a way so consistency and clarity will go a long way in readability.

Reply:
Good catch, thanks!